neuro_flow/parser.py

```diff
@@ -837,28 +874,46 @@ def parse_flow_main(ctor: BaseConstructor, node: yaml.MappingNode) -> ast.BaseFl
 FlowLoader.add_constructor("flow:main", parse_flow_main)  # type: ignore
 
 
-def parse_live(workspace: LocalPath, config_file: LocalPath) -> ast.LiveFlow:
+def parse_live(
+    workspace: LocalPath, config_file: LocalPath
+) -> Tuple[ast.LiveFlow, Digest]:
     # Parse live flow config file
+    hasher = hashlib.new("sha256")
+    with config_file.open("rb") as bf:
```

Comment:
Minor: you can move "create digest for file" logic to special function to avoid duplication.

Reply:
Ideally, both hash calculation and yaml construction should be done in the single iteration over the file.
It is not done yet but I'd like to keep this bad-smelling code as is until the future improvement.
Perhaps, it will be a custom `yaml.Reader` instance.